### PR TITLE
BREAKING: move `std/flags` to new `std/cli` sub-module

### LINKED ISSUE #3526: proposal: Move `std/flags` to `std/cli`
Currently, we have this top level module which is both confusingly named (what is flags referring to? -- not immediately clear imo) and incredibly limiting in scope. I think this would be a lot better if the TLM was named `std/cli`. This would also open a lot of opportunities for new utilities that I would find quite useful imo.

See https://github.com/denoland/deno_std/pull/3488 for context on initial discussion.

cc @jeff-hykin @kt3k @iuioiua

### PR DESCRIPTION
**Pull request by lino-levan**
Closes #3526. Marking as draft to prevent accidental merge before discussion is complete.

### CODE REVIEW

File: cli/flags.ts
Comment (iuioiua) [inline cli/flags.ts:1]: I think we should name this file `parse_flags.ts` and the function `parseFile()`. WDYT?

File: cli/mod.ts
Comment (iuioiua) [inline cli/mod.ts:1]: Please don't forget to include the JSDocs from `flags/mod.ts`.

File: flags/mod.ts
Comment (iuioiua) [inline flags/mod.ts:5]: Perhaps, we're okay to remove after 0.208.0. @kt3k, WDYT?

**Review by iuioiua — CHANGES_REQUESTED** [review]
Let's go ahead with this move.

File: cli/flags.ts
Comment (lino-levan) [reply cli/flags.ts:1]: I'd made rename it to `parse_args.ts` and `parseArgs()`.

Comment (nhrones) [reply cli/flags.ts:1]: > I would rename it to `parse_args.ts` and `parseArgs()` if it was up to me.

In my very ancient experience.  Flags were almost always Boolean indicators.  Often masking bits to pack flags.
Of course that just one old mans take on naming.

**Comment by nhrones** [comment]
https://en.wikipedia.org/wiki/Flag_(programming)

File: cli/flags.ts
Comment (iuioiua) [reply cli/flags.ts:1]: Actually, it's parsing `Deno.args()`. My bad. Yep, let's go with `parseArgs()`.

File: cli/mod.ts
Comment (iuioiua) [inline cli/mod.ts:4]: How about a short description for this sub-module? Maybe something like "Tools for interacting with the Deno CLI."

File: flags/mod.ts
Comment (iuioiua) [inline flags/mod.ts:5]: I think we should punt the removal version back to 0.209.0 now.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Looking good! Couple nits.

File: flags/mod.ts
Comment (kt3k) [reply flags/mod.ts]: `std/flags` exists for long time and is heavily used. I think we shouldn't remove this that soon

Comment (lino-levan) [reply flags/mod.ts]: Is `1.0.0` too late or what do you think about the timing?

Comment (kt3k) [reply flags/mod.ts]: I think removal at 1.0.0 is safer and more appropriate regarding the usage of this module.

File: cli/mod.ts
Comment (iuioiua) [reply cli/mod.ts:4]: Happy to approve once this is addressed, @lino-levan.

**Review by iuioiua — APPROVED** [review]
LGTM. Thank you, Lino.

**Comment by lino-levan** [comment]
wait, don't merge yet

**Comment by lino-levan** [comment]
Alright, should be mergeable now.

**Comment by iuioiua** [comment]
> wait, don't merge yet

All good. I don't merge these more important PRs without Bartek's or Yoshiya's approval anyway 😛

**Review by kt3k — APPROVED** [review]
LGTM

The core team is also in favor of `std/cli` idea.

**Pull request merged**
